pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,325 @@
+	envs := []corev1.EnvVar{
+		{
+			Name:  "CLUSTER_NAME",
+			Value: pr.cfg.ESClusterConfig.ClusterName(),
+		},
+		{
+			Name:  "MULTI_CLUSTER_FORWARDING_CA",
+			Value: pr.cfg.TrustedBundle.MountPath(),
+		},
+		{
+			Name:  "FIPS_MODE_ENABLED",
+			Value: operatorv1.IsFIPSModeEnabledString(pr.cfg.Installation.FIPSMode),
+		},
+	}
+
+	sc := securitycontext.NewNonRootContext()
+
+	volumeMounts := []corev1.VolumeMount{
+		pr.cfg.TrustedBundle.VolumeMount(pr.SupportedOSType()),
+		pr.cfg.PolicyRecommendationServerCertSecret.VolumeMount(pr.SupportedOSType()),
+	}
+
+	return corev1.Container{
+		Name:            "policy-recommendation-controller",
+		Image:           pr.image,
+		Env:             envs,
+		SecurityContext: sc,
+		VolumeMounts:    volumeMounts,
+	}
+}
+
+func (pr *policyRecommendationComponent) deployment() *appsv1.Deployment {
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PolicyRecommendationName,
+			Namespace: PolicyRecommendationNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.Int32ToPtr(1),
+			Template: *pr.deploymentPodTemplate(),
+		},
+	}
+}
+
+func (pr *policyRecommendationComponent) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range pr.cfg.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+
+	volumes := []corev1.Volume{
+		pr.cfg.TrustedBundle.Volume(),
+		pr.cfg.PolicyRecommendationServerCertSecret.Volume(),
+	}
+
+	container := relasticsearch.ContainerDecorateIndexCreator(
+		relasticsearch.ContainerDecorate(pr.controllerContainer(), pr.cfg.ESClusterConfig.ClusterName(),
+			ElasticsearchPolicyRecommendationUserSecret, pr.cfg.ClusterDomain, rmeta.OSTypeLinux),
+		pr.cfg.ESClusterConfig.Replicas(), pr.cfg.ESClusterConfig.Shards())
+
+	return relasticsearch.DecorateAnnotations(&corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        PolicyRecommendationName,
+			Namespace:   PolicyRecommendationNamespace,
+			Annotations: pr.policyRecommendationAnnotations(),
+		},
+		Spec: corev1.PodSpec{
+			Tolerations:        pr.cfg.Installation.ControlPlaneTolerations,
+			NodeSelector:       pr.cfg.Installation.ControlPlaneNodeSelector,
+			ServiceAccountName: PolicyRecommendationName,
+			ImagePullSecrets:   ps,
+			Containers: []corev1.Container{
+				container,
+			},
+			Volumes: volumes,
+		},
+	}, pr.cfg.ESClusterConfig, pr.cfg.ESSecrets).(*corev1.PodTemplateSpec)
+}
+
+func (pr *policyRecommendationComponent) policyRecommendationAnnotations() map[string]string {
+	annotations := pr.cfg.TrustedBundle.HashAnnotations()
+
+	pr.cfg.TrustedBundle.HashAnnotations()
+	if pr.cfg.PolicyRecommendationServerCertSecret != nil {
+		annotations[pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationKey()] = pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationValue()
```

Comment:
what about the trusted bundle? Do we need to reboot if that changes?

Reply:
It's currently what we do for all components, but it should only change when our CA expires or if users bring their own tls secrets, which is not a very common thing to do. Especially for the items in this bundle.